nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/engine/activityimpl/ActivityDef.java

```diff
@@ -230,4 +242,65 @@ public NBLabels auxLabels() {
         return NBLabels.forKV();
 
     }
+
+    public NBConfigModel getConfigModel() {
+        ConfigModel cfgmodel = ConfigModel.of(this.getClass());
+        Map<String, String> params = parameterMap.getStringStringMap();
+        params.forEach((k, v) -> {
+            cfgmodel.add(Param.defaultTo(k, v, "activity parameter found on command line"));
+        });
+        if ( ! params.containsKey(FIELD_ALIAS)) {
```

Comment:
I'm not entirely sure what this section should be doing with all the conditionals. This is what I'd expect: 1) individual config models are declarative and definitive about something specific, so no conditions are needed on a per-param basis. 2) config models can be derived by disjunction or intersection with others. 3) defaults in the config model are manifested when the config model is asked for the parameter that supports it.

But I might be missing something here. Can you elaborate a bit on what the purpose of this section is?

Reply:
`ConfigModel.add` adds keys to a model regardless of whether or not the value has already been added to the model. If the field was already by the `params.forEach` loop then it is skipped. A better fix might be to change `ConfigModel.add` behavior.

The usefulness of having a ConfigModel for ActivityDef is still speculative, but will be more for checking error conditions with excess.

Reply:
I've adjusted the behavior in `ConfigModel.add`